MyOnboarding/test/MyOnboardingApp.Api.Tests/TodoListControllerTest.cs

```diff
@@ -1,55 +1,85 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.Routing;
 using MyOnboardingApp.Api.Controllers;
-using MyOnboardingApp.Api.Models;
+using MyOnboardingApp.Contracts.Models;
+using MyOnboardingApp.Contracts.Repository;
+using MyOnboardingApp.Contracts.UrlLocation;
 using MyOnboardingApp.Tests.Utils;
+using NSubstitute;
 using NUnit.Framework;
 
 namespace MyOnboardingApp.Tests
 {
     [TestFixture]
     public class TodoListControllerTest
     {
-        private readonly TodoListController _controller = new TodoListController();
+        private ITodoListRepository _repository; 
         private readonly Guid _expectedId = Guid.Empty;
+        private TodoListController _controller;
+        private IUrlLocator _itemUrlLocator;
+        private IEqualityComparer<TodoListItem> _comparer;
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            _itemUrlLocator = Substitute.For<IUrlLocator>();
+            _repository = Substitute.For<ITodoListRepository>();
+            _controller = new TodoListController(_repository, _itemUrlLocator)
+            {
+                Request = new HttpRequestMessage(),
+                Configuration = new HttpConfiguration(),
+                Url = new UrlHelper()
+            };
+            _comparer = new ItemEqualityComparer();
+            await Task.CompletedTask;
+        }
 
         [Test]
         public async Task Get_NoIdSpecified_ReturnsCorrectResponse()
         {
-            var msg = await _controller.GetMessageFromAction(controller => controller.GetAsync());
+            var msg = await _controller.GetMessageFromAction(control => control.GetAsync());
```

Comment:
I'm sure that this [`monosodium glutamate`](https://www.urbandictionary.com/define.php?term=MSG) is crucial for the test 😆 → Don't use abbreviations (this applies to `controller` renamed to `control` as well 😉 )